File: Package/VB/Client.AE/ObservedNode.vb
' Copyright (c) Traeger Industry Components GmbH. All Rights Reserved.

Imports System

Imports Opc.UaFx
Imports Opc.UaFx.Client

Namespace AE
    Friend Class ObservedNode
        Public Sub New(parentNodeId As OpcNodeId, name As String)
            Me.Id = OpcNodeId.Of(name, parentNodeId)
            Me.Label = (name + ":").PadRight(16)
        End Sub

        Public ReadOnly Property Id As OpcNodeId
        Public Property EventData As OpcEvent
        Public Property Label As String
        Public Property Unit As OpcEngineeringUnitInfo
        Public Property Value As OpcValue


        Public Sub Initialize(client As OpcClient)
            Dim node = client.BrowseNode(Me.Id)
            Dim analogNode = TryCast(node, OpcAnalogItemNodeInfo)

            If (Not analogNode Is Nothing) Then
                Me.Unit = analogNode.EngineeringUnit
            End If

            Me.Value = client.ReadNode(Me.Id)
        End Sub
    End Class
End Namespace

File: cs/Basic/Client.AE/ObservedNode.cs
// Copyright (c) Traeger Industry Components GmbH. All Rights Reserved.

namespace AE
{
    using Opc.UaFx;
    using Opc.UaFx.Client;

    internal class ObservedNode
    {
        public ObservedNode(OpcNodeId parentNodeId, string name)
            : base()
        {
            this.Id = OpcNodeId.Of(name, parentNodeId);
            this.Label = (name + ":").PadRight(16);
        }

        public OpcNodeId Id
        {
            get;
        }

        public OpcEvent Event
        {
            get;
            set;
        }
        
        public string Label
        {
            get;
        }

        public OpcEngineeringUnitInfo Unit
        {
            get;
            private set;
        }

        public OpcValue Value
        {
            get;
            set;
        }


        public void Initialize(OpcClient client)
        {
            var node = client.BrowseNode(this.Id);
            var analogNode = node as OpcAnalogItemNodeInfo;

            if (analogNode != null)
                this.Unit = analogNode.EngineeringUnit;

            this.Value = client.ReadNode(this.Id);
        }
    }
}
